src/lib.rs

```diff
@@ -5,22 +5,80 @@
 
 #![allow(dead_code)]
 
-pub mod example_result;
-pub mod events;
-pub mod context;
+#[macro_use]
+extern crate derive_builder;
```

Comment:
we may want to use [derive_new](https://github.com/nrc/derive-new) for all the `fn new()` boilerplate here and there

Reply:
I have marked the opportunities